app/src/main/java/com/example/drawingApp/DialogObject.kt

```diff
@@ -0,0 +1,127 @@
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogObject {
+
+
+    fun showLayerAlertDialog(
+        c: Context,
+        viewModel: DrawingViewModel,
+        /**
+         *Called when the alert Dialog is submitted
+         * Determines what to do with the final sanitized string
+         * current implementations are for adding to the list of layers
+         * and for editing a layer
+         */
+        onDialogSubmission: (String) -> Unit
+    ) {
+        val inflater = LayoutInflater.from(c)
+        val view = inflater.inflate(R.layout.layer_dialog, null)
+        val layerField = view.findViewById<EditText>(R.id.newLayerField)
+        val errorText = view.findViewById<TextView>(R.id.dialogError)
+        val layerNames = mutableListOf<String>()
+        for (x in viewModel.layerNames) layerNames.add(x.text)
+        var layerCounter = 1
+
+
+        //find the first available layer name not taken
+        while (layerNames.contains(
+                c.resources.getString(
+                    R.string.new_layer_hint,
+                    layerCounter
+                )
+            )
+        ) {
+            layerCounter += 1
+        }
+        layerField.setText(c.resources.getString(R.string.new_layer_hint, layerCounter))
+
+        //this text changed listener needs to be after the set text
+        //checks validity of the string length in the layer field, if not valid shows an error
+        layerField.addTextChangedListener {
+            val fieldString = it.toString().trim()
+            viewModel.currentLayerCreationDialogText = fieldString
+            errorText.setText(R.string.charCountError)
+            errorText.isVisible = fieldString.length > MainActivity.MAX_CHAR_LEN
+        }
+
+        val dialog = AlertDialog.Builder(c).apply {
+            setView(view)
+        }.create()
+
+        view.findViewById<TextView>(R.id.newLayerCancel).setOnClickListener {
+            dialog.dismiss()
+        }
+
+        view.findViewById<TextView>(R.id.newLayerOk).setOnClickListener {
+            val fieldString = layerField.text.toString().trim()
+            if (fieldString.length > MainActivity.MAX_CHAR_LEN) {
+                errorText.visibility = View.VISIBLE
+            } else if (layerNames.contains(fieldString)) {
+                errorText.setText(R.string.layerExistsError)
+                errorText.visibility = View.VISIBLE
+            } else {
+                layerNames.add(fieldString)
+                onDialogSubmission(fieldString)
+                //model.layerNames.add(LayerViewModel(fieldString))
+                Toast.makeText(c, fieldString, Toast.LENGTH_SHORT)
+                    .show()
+                dialog.dismiss()
+            }
+        }
+        //this is here so that in the case that the dialogue was previously open
```

Comment:
`Dialog` is for software components. `Dialogue` is for conversational words in a story